### DISCUSSION TITLE
loop condition

### DISCUSSION THREAD

**Discussion by ghost** [john-science/mazelib#147] [Q&A] [answered]
May I ask why here is `< self.h * self.w`, it seems this loop will never end? Sorry I didn't run the code.
https://github.com/john-science/mazelib/blob/9b4190f39c1c939726b3a0b33284cb164f31d43a/mazelib/generate/AldousBroder.py#L37

**Comment by john-science** [comment, marked as answer]
The reason that loop is valid is because a few lines lower there is this:

https://github.com/john-science/mazelib/blob/9b4190f39c1c939726b3a0b33284cb164f31d43a/mazelib/generate/AldousBroder.py#L55

So, yeah, the code runs and mazes are correctly generated.
